### PR TITLE
Add GO GraphQL Server

### PR DESCRIPTION
**Pull request by rawagner**
Another variant to Apollo Server https://github.com/openshift/console/pull/5215 . This time server is implemented in GO and client is still Apollo with one Websocket connection.

Implementation is based on [Gophers GraphQL](https://github.com/graph-gophers/graphql-go) - this library has a [wrapper](https://github.com/graph-gophers/graphql-transport-ws) for default Relay client to make it work with Apollo WS.

Right now the GraphQL server is part of the Bridge process, not a standalone pod/sidecar container. There's no need to modify the existing deployment process of console.

I've hit a few issues with the [wrapper](https://github.com/graph-gophers/graphql-transport-ws) lib so for now I've used my fork to workaround those.
Issues are:
1. queries are not executed in parallel - PR https://github.com/graph-gophers/graphql-transport-ws/pull/15
2. cannot pass custom context to GraphQL resolvers - no PR yet, but fixed in my fork

@spadgett @benjaminapetersen @christianvogt @vojtechszocs @jelkosz

### CODE REVIEW

**Comment by rawagner** [comment]
**Updates**

**Backend:**

- I've been able to reuse the types from k8s.io/api package so we can easily convert the json response to golang struct, but we still need to define types for GraphQL schema. Good thing is that there's type checking between GraphQL schema types and return values of resolvers.

- I've also added passthrough for Impresonate headers 

**Frontend:**

- Added TS types generation from GraphQL schema and GraphQL queries. We can use these generated types to properly type the Apollo client

- Added eslint plugin which checks the validity of GraphQL queries against the GraphQL schema.

- GraphQL queries are now defined in separate (.gql, .graphql) files. These files are preprocessed into ASTs during the webpack build which will result in better performance.

I think that we are pretty well covered in terms of type safety for both backend and frontend. 
@spadgett

File: frontend/@types/queries.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+declare module '*/features.gql' {
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/@types/queries.d.ts]: we probably dont need them, added to `.gitignore`

File: pkg/graphql/resolver/query.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+				if value != "" {
+					request.Header.Add(key, value)
+				}
+			}
+		}
+	}
+}
+
+func (r *QueryResolver) FetchURL(ctx context.Context, args struct{ URL string }) (string, error) {
+	request, _ := http.NewRequest("GET", args.URL, nil)
```
[the remark this answers is not in the document]
Comment (rawagner) [reply pkg/graphql/resolver/query.go]: error handling added

File: pkg/graphql/resolver/query.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+func (r *QueryResolver) SelfSubjectAccessReview(ctx context.Context, args SSARArgs) (*auth.SelfSubjectAccessReview, error) {
+	spec := auth.SelfSubjectAccessReview{
+		Spec: auth.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: parseArgs(args),
+		},
+	}
+	buf := new(bytes.Buffer)
+	json.NewEncoder(buf).Encode(&spec)
+	request, _ := http.NewRequest("POST", "/apis/"+auth.SchemeGroupVersion.String()+"/selfsubjectaccessreviews", buf)
```
[the remark this answers is not in the document]
Comment (rawagner) [reply pkg/graphql/resolver/query.go]: error handling added

File: frontend/codegen.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+schema: ../pkg/graphql/schema.graphql
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/codegen.yml:1]: Didnt before, but its now part of `build`, `dev` and `dev-once` scripts

**Comment by rawagner** [comment]
> What work do you feel like is remaining?

I'm quite happy with how things look right now and the basic functionality provided is enough for first PR. I don't plan to do any more work (besides addressing any other feedback that you have).

In the following PRs I'd like to implement some subscription and prometheus querying.

**Comment by spadgett** [comment]
Looks like we'll need to update `MAX_BYTES` in analyze.sh due to apollo-client. Happy to make this trade-off since this PR improves page load time, which is what the test is concerned with.

```
Main vendor bundle size: 2.52 MiB
FAILURE: Main vendor bundle is larger than the 2.5 MiB limit.
```

**Comment by rawagner** [comment]
@spadgett I've updated limit to 2.6MiB, hope thats ok.

**Comment by rawagner** [comment]
rebased on latest master

File: frontend/package.json [hunk 5]
```diff
@@ -28,15 +28,18 @@
     "test-suite": "ts-node -O '{\"module\":\"commonjs\"}' ./node_modules/.bin/protractor integration-tests/protractor.conf.ts",
     "debug-test-suite": "TS_NODE_COMPILER_OPTIONS='{\"module\":\"commonjs\"}' node -r ts-node/register --inspect-brk ./node_modules/.bin/protractor integration-tests/protractor.conf.ts",
     "analyze": "NODE_ENV=production NODE_OPTIONS=--max-old-space-size=4096 ts-node -O '{\"module\":\"commonjs\"}' ./node_modules/.bin/webpack --mode=production --profile --json | awk '{if(NR>2)print}' > public/dist/stats.json && ts-node -O '{\"module\":\"commonjs\"}' ./node_modules/.bin/webpack-bundle-analyzer --mode static -r public/dist/report.html public/dist/stats.json",
-    "prettier-all": "prettier --write '**/*.{js,jsx,ts,tsx,json}'"
+    "prettier-all": "prettier --write '**/*.{js,jsx,ts,tsx,json}'",
+    "gql-generate": "graphql-codegen"
```
Comment (spadgett) [inline frontend/package.json:32]: Is this just an alias? Any reason not to use `yarn run graphql-codegen` directly?

**Comment by stlaz** [comment]
@rawagner @spadgett FYI - this PR appear to have broken release jobs for 4.6, please investigate how it passed CI checks (although as far as I can see the commit shows it didn't).

**Comment by rawagner** [comment]
@stlaz I guess its because I didnt include `schema.graphql` file to `Dockerfile.product` . Fix is here https://github.com/openshift/console/pull/5843

**Pull request merged**